processing/src/main/java/io/druid/query/topn/BaseTopNAlgorithm.java

```diff
@@ -101,6 +104,8 @@ public void run(
 
   protected abstract DimValSelector makeDimValSelector(Parameters params, int numProcessed, int numToProcess);
 
+  protected abstract int computeNewLength(DimValSelector dimValSelector, int numProcessed, int numToProcess);
```

Comment:
we should add javadocs here to explain what this method does and what the parameters mean